presto-hive/src/test/java/com/facebook/presto/hive/util/TestHiveFileIterator.java

```diff
@@ -0,0 +1,229 @@
+                updateExpectedStats(RECURSE, true, 5);
+                updateExpectedStats(RECURSE, false, 5);
+                List<File> subDirs = createDirs(dirs.get(i).getAbsolutePath(), 1);
+                createFiles(subDirs.get(0).getAbsolutePath(), 5, true, null);
+                updateExpectedStats(RECURSE, false, 5);
+            }
+        }
+        hadoopConf = new Configuration();
+        hadoopConf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
+        hadoopConf.set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem.class.getName());
+
+        listDirectoryOperation = path ->
+        {
+            FileSystem fs = path.getFileSystem(hadoopConf);
+            return fs.listLocatedStatus(path);
+        };
+    }
+
+    private void createFiles(String basePath, int numFiles, boolean randomName, String fileNamePrefix) throws IOException
+    {
+        for (int i = 0; i < numFiles; i++) {
+            new File(basePath).mkdirs();
+            String fileName;
+            if (randomName) {
+                fileName = getRandomStringOfLen(10);
+            }
+            else {
+                fileName = fileNamePrefix + "_" + i;
+            }
+            new File(basePath + File.separator + fileName).createNewFile();
+        }
+    }
+
+    private List<File> createDirs(String basePath, int numDirectories)
+    {
+        List<File> directories = new ArrayList<>();
+        for (int i = 0; i < numDirectories; i++) {
+            String dirName = basePath + File.separator + pathFilterString + "_" + getRandomStringOfLen(5);
+            File file = new File(dirName);
+            file.mkdirs();
+            directories.add(file);
+        }
+        return directories;
+    }
+
+    private String getRandomStringOfLen(int len)
+    {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < len; i++) {
+            sb.append(RANDOM_FILE_NAME_SALT_STRING.charAt(random.nextInt(RANDOM_FILE_NAME_SALT_STRING.length())));
+        }
+        return sb.toString();
+    }
+
+    private void updateExpectedStats(HiveFileIterator.NestedDirectoryPolicy policy, boolean hasPathFilter, int expectedCount)
+    {
+        switch (policy) {
+            case IGNORED:
+                if (hasPathFilter) {
+                    fileCountForIgnoredPolicyPathFilterOn += expectedCount;
+                }
+                else {
+                    fileCountForIgnoredPolicyPathFilterOff += expectedCount;
+                }
+                break;
+            case RECURSE:
+                if (hasPathFilter) {
+                    fileCountForRecursePolicyPathFilterOn += expectedCount;
+                }
+                else {
+                    fileCountForRecursePolicyPathFilterOff += expectedCount;
+                }
+                break;
+        }
+    }
+
+    public int iteratorTestHelper(HiveFileIterator hiveFileIterator)
```

Comment:
nit: can we have a more descriptive name for this method? like `getFileCount` (maybe?)

Reply:
will rename.